vpt/processors/gaze_detection/__init__.py

```diff
@@ -0,0 +1,226 @@
+        top_y = box[1] + offset[1]
+        right_x = box[2] + offset[0]
+        bottom_y = box[3] + offset[1]
+        return [left_x, top_y, right_x, bottom_y]
+
+    @staticmethod
+    def get_square_box(box):
+        """Get a square box out of the given box, by expanding it."""
+        left_x = box[0]
+        top_y = box[1]
+        right_x = box[2]
+        bottom_y = box[3]
+
+        box_width = right_x - left_x
+        box_height = bottom_y - top_y
+
+        # Check if box is already a square. If not, make it a square.
+        diff = box_height - box_width
+        delta = int(abs(diff) / 2)
+
+        if diff == 0:  # Already a square.
+            return box
+        elif diff > 0:  # Height > width, a slim box.
+            left_x -= delta
+            right_x += delta
+            if diff % 2 == 1:
+                right_x += 1
+        else:  # Width > height, a short box.
+            top_y -= delta
+            bottom_y += delta
+            if diff % 2 == 1:
+                bottom_y += 1
+
+        # Make sure box is always square.
+        assert ((right_x - left_x) == (bottom_y - top_y)), 'Box is not square.'
+
+        return [left_x, top_y, right_x, bottom_y]
+
+    @staticmethod
+    def box_in_image(box, image):
+        """Check if the box is in image"""
+        rows = image.shape[0]
+        cols = image.shape[1]
+        return box[0] >= 0 and box[1] >= 0 and box[2] <= cols and box[3] <= rows
+
+    def extract_cnn_facebox(self, image):
+        """Extract face area from image."""
+        _, raw_boxes = self.face_detector.get_faceboxes(
+            image=image, threshold=0.5)
+        a = []
+        for box in raw_boxes:
+            # Move box down.
+            # diff_height_width = (box[3] - box[1]) - (box[2] - box[0])
+            offset_y = int(abs((box[3] - box[1]) * 0.1))
+            box_moved = self.move_box(box, [0, offset_y])
+
+            # Make box square.
+            facebox = self.get_square_box(box_moved)
+
+            if self.box_in_image(facebox, image):
+                a.append(facebox)
+
+        return a
+
+    def detect_marks(self, image_np):
+        """Detect marks from image"""
+
+        # # Actual detection.
+        predictions = self.model.signatures["predict"](
+            tf.constant(image_np, dtype=tf.uint8))
+
+        # Convert predictions to landmarks.
+        marks = np.array(predictions['output']).flatten()[:136]
+        marks = np.reshape(marks, (-1, 2))
+
+        return marks
+
+
+mark_detector = MarkDetector()
+font = cv2.FONT_HERSHEY_SIMPLEX
+# 3D model points.
+model_points = np.array([
+    (0.0, 0.0, 0.0),  # Nose tip
+    (0.0, -330.0, -65.0),  # Chin
+    (-225.0, 170.0, -135.0),  # Left eye left corner
+    (225.0, 170.0, -135.0),  # Right eye right corne
+    (-150.0, -150.0, -125.0),  # Left Mouth corner
+    (150.0, -150.0, -125.0)  # Right mouth corner
+])
+
+
+class GazeDetection(ProcessorBase[np.ndarray]):
+    subj = Subject()
```

Comment:
This declares `subj` as a class-level (static) member. Move this to `__init__` to make it an instance field